File: frontend/streamlit_app/services/api_client.py
import os

import httpx
from dotenv import load_dotenv

load_dotenv()

API_BASE_URL = os.getenv("API_BASE_URL", "http://localhost:8000")
TIMEOUT = 120.0


def analyze(raw_text: str, url: str | None = None, title: str | None = None) -> dict:
    """POST to /api/v1/analyze and return the parsed JSON response."""
    payload = {"raw_text": raw_text, "url": url, "title": title}
    with httpx.Client(timeout=TIMEOUT) as client:
        response = client.post(f"{API_BASE_URL}/api/v1/analyze", json=payload)
        response.raise_for_status()
        return response.json()


def health() -> dict:
    """GET /health and return parsed JSON."""
    with httpx.Client(timeout=10.0) as client:
        response = client.get(f"{API_BASE_URL}/health")
        response.raise_for_status()
        return response.json()

File: frontend/extension/src/lib/api_client.js
async function getApiBaseUrl() {
  return new Promise((resolve) => {
    chrome.storage.local.get("apiBaseUrl", (data) => {
      resolve(data.apiBaseUrl || "https://backend-production-1d50.up.railway.app");
    });
  });
}

export async function analyzeText(rawText, url = null, title = null) {
  const baseUrl = await getApiBaseUrl();

  const submitRes = await fetch(`${baseUrl}/api/v1/analyze`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ raw_text: rawText, url, title }),
  });

  if (!submitRes.ok) {
    const err = await submitRes.text();
    throw new Error(`API error ${submitRes.status}: ${err}`);
  }

  const { job_id } = await submitRes.json();

  while (true) {
    await new Promise((resolve) => setTimeout(resolve, 5000));

    const pollRes = await fetch(`${baseUrl}/api/v1/analyze/${job_id}`);
    if (!pollRes.ok) throw new Error(`Poll error ${pollRes.status}`);

    const job = await pollRes.json();
    if (job.status === "done") return job.result;
    if (job.status === "error") throw new Error(job.error || "Analysis failed");
  }
}
